File: Sorting Algorithms/Bubble.cpp
#include<iostream>
using namespace std;

//Bubble Sort
void BubbleSort(int array[],int size){
    //The First Loop is For the Number of repetition
    for(int step=0;step<size;step++) {
    
    //This loop for comparing and swaping in Each Repetition.
     for(int i=0;i<size-step;i++) {
            
            if(array[i]>array[i+1]) {
                int temp=array[i];
                array[i]=array[i+1];
                array[i+1]=temp;
            }
        }
    }
}

void printArray(int array[],int size){
    for(int i=0;i<size;i++){
        cout<<array[i]<<" , ";
    }
    cout<<endl;
}

int main(){
    int arr[]={5,4,3,2,1};
    int size=sizeof(arr)/sizeof(arr[0]);

    printArray(arr,size);

    BubbleSort(arr,size);

    printArray(arr,size);
}

File: Sorting Algorithms/Bubble.py
#Bubble Sort for Sorting in ascending Order
def BubbleSort(arr):
    #this loop is to iterate through each element
    for step in range(len(arr)):

        #this loop is to compare each element
        for i in range(len(arr)-step-1):
            #compare adjacent element
            #swap if left element is greater then right element.
            if arr[i]>arr[i+1]:
                arr[i],arr[i+1]=arr[i+1],arr[i] 
                
arr=[5,4,3,2,1]

print(arr)

BubbleSort(arr)

print(arr)
